### PR TITLE
kubeval 0.16.1 (new formula)

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

previous effort, #17277
relates to instrumenta/kubeval#26

### CODE REVIEW

File: Formula/kubeval.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  license "Apache-2.0"
+  head "https://github.com/instrumenta/kubeval.git", branch: "master"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{tap.user.downcase}
```
Comment (dawidd6) [inline Formula/kubeval.rb:15]: Is this correct?

File: Formula/kubeval.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      -X main.date=#{time.iso8601}
+    ].join(" ")
+    system "go", "build", *std_go_args(ldflags: ldflags)
+
+    pkgshare.install "fixtures"
```
Comment (dawidd6) [inline Formula/kubeval.rb:20]: If those are only used for our test then I would say we should create a resource for this purpose.

File: Formula/kubeval.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/kubeval.rb:15]: yeah, it was kind of intentional. But I can switch `tag-revision` checkout.

File: Formula/kubeval.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/kubeval.rb:20]: Yeah, that also works for me, but in general, I do like to have fixtures as examples in the `pkgshare`. What do you think?

Comment (dawidd6) [reply Formula/kubeval.rb:20]: Okay I see some formulae with logic similar to this.

File: Formula/kubeval.rb (hunk 1 above)
Comment (dawidd6) [reply Formula/kubeval.rb:15]: Yes, tag-revision seems to be more appropriate in this case.

Comment (chenrui333) [reply Formula/kubeval.rb:15]: cool, let me switch to `tag-revision` checkout.

**Review by dawidd6 — DISMISSED** [review]

File: Formula/kubeval.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  test do
+    system bin/"kubeval", pkgshare/"fixtures/valid.yaml"
+    assert_equal 0, $CHILD_STATUS.exitstatus
```
Comment (SMillerDev) [inline Formula/kubeval.rb:26]: Is there any output here?

Comment (chenrui333) [reply Formula/kubeval.rb:26]: I dont see any output, that is why I assert the `valid` config check status.

Comment (dawidd6) [reply Formula/kubeval.rb:26]: I believe `system` call will fail the test if the command returns non-zero status, so this line might not even be needed.

Comment (chenrui333) [reply Formula/kubeval.rb]: That sounds about right, just removed the return status check.

**Review by branchv — APPROVED** [review]

**Pull request closed without merging**